gridstatus/ercot.py

```diff
@@ -2939,13 +2950,29 @@ def _get_documents(
             except Exception:
                 friendly_name_timestamp = None
 
+            friendly_name = doc["Document"]["FriendlyName"]
+
+            # ERCOT adds xhr to the file name during the repeated hour that occurs
+            # during DST end. However, ERCOT may get the timezone offset wrong.
```

Comment:
I believe the `PublishDate` offset from ERCOT is wrong during DST and should be ignored (as implemented in this PR). This would mean that our publish time during DST has been off by one hour for ERCOT.

Reply:
> ERCOT may get the timezone offset wrong

I'm confused here. ERCOT is showing a timezone offset of 6 hours for the publish time even when DST is in effect. This seems wrong. The offset should be 5 hours during DST and 6 hours outside of DST.

<img width="804" alt="Screenshot 2024-11-03 at 03 40 55" src="https://github.com/user-attachments/assets/efb8b3be-b622-4678-8b98-b296cc46670e">

Reply:
The Posted Time is showing as 2024-11-02 9:30 PM which should have an offset of 5 hours. We've just been parsing the `PublishDate` as given.

Reply:
> This would mean that our publish time during DST has been off by one hour for ERCOT.

does this apply to our history or just on DST transition days?

Reply:
I'm not sure. Looking at past reports shows the same incorrect `PublishDate` offset but ERCOT may be applying the non-DST offset to all timestamps it displays. now that DST is over. In other words, if we looked at those publish times yesterday, it's possible we would have seen the correct offset.

<img width="818" alt="Screenshot 2024-11-03 at 06 33 48" src="https://github.com/user-attachments/assets/1b00a714-c30e-46a7-8138-9bb46ed5f263">

Reply:
confusing. Let me look at this a bit more closely before we merge

Reply:
in reviewing the [list documents from the report](https://www.ercot.com/misapp/servlets/IceDocListJsonWS?reportTypeId=12311) in your screenshot and then also the [Real Time Settlement Point](https://www.ercot.com/misapp/servlets/IceDocListJsonWS?reportTypeId=12301&_=1730641843741) list, I reached the same conclusion with you and agree with this fix

Just to confirm my logic:
1. Ercot is using the UTC offset at the time document list is returned and adds it to the all publish times 
2. However, ERCOT doesn't adjust the time of timestamp, it just add the offset. 
3. This leads to situations where the publish time indicates a `-6` offset, but it should be interpreted as `-5`
4. Therefore, we should ignore the ERCOT offset and use `xhr` in the file name to differentiate when the time stamp is ambiguous 

My only concern here is does this change work across the datasets we've implemented that call `_get_documents`

Reply:
Your logic matches my conclusion. We will have to wait until DST starts again though to confirm.

> My only concern here is does this change work across the datasets we've implemented that call _get_documents

ERCOT says that "xhr" in a file name represents the second file published in a repeated hour.

> ERCOT has appended an xhr indicator in the file naming convention to create file uniqueness across the repeated hour files generated. The first file generated will post with the expected naming convention. The second file generated will appear with the _xhr designation.

https://lists.ercot.com/cgi-bin/wa?A3=1111&L=NOTICE_TRAINING&E=quoted-printable&P=4519&B=--_000_B117FDA9B7BC68479362C1197F77D8790950ADCPW0005ercotcom_&T=text%2Fhtml;%20charset=us-ascii&XSS=3&header=1